integration/benchmark/worker_stats_tracker.go

```diff
@@ -81,47 +114,26 @@ func (st *WorkerStatsTracker) GetTxSent() int {
 	return st.stats.txsSent
 }
 
-func (st *WorkerStatsTracker) GetStats() WorkerStats {
+func (st *WorkerStatsTracker) getStats() workerStats {
 	st.mux.Lock()
 	defer st.mux.Unlock()
 
+	st.stats.txsSentEWMA = st.txsSentEWMA.Value()
+	st.stats.txsExecutedEWMA = st.txsExecutedEWMA.Value()
 	return st.stats
 }
 
-// AvgTPSBetween returns the average transactions per second TPS between the two time points
-func (st *WorkerStatsTracker) AvgTPSBetween(start, stop time.Time) float64 {
-	sum := 0
-
-	st.mux.Lock()
-	defer st.mux.Unlock()
-	for timestamp, count := range st.txsSentPerSecond {
-		if timestamp < start.Unix() || timestamp > stop.Unix() {
-			continue
-		}
-		sum += count
-	}
-
-	diff := stop.Sub(start)
-
-	return float64(sum) / diff.Seconds()
-}
-
-func (st *WorkerStatsTracker) Digest() string {
-	t := table.NewWriter()
-	t.AppendHeader(table.Row{
-		"workers",
-		"total TXs sent",
-		"total TXs finished",
-		"Avg TPS (last 10s)",
+func NewPeriodicStatsLogger(st *WorkerStatsTracker, log zerolog.Logger) *Worker {
+	w := NewWorker(0, 1*time.Second, func(workerID int) {
```

Comment:
aside: use the main's ctx rather than Background to create the worker's internal ctx.